service/worker/scanner/README.md

```diff
@@ -0,0 +1,247 @@
+# Scanner / fixer / etc how-to
+
+For future scanner-extenders/rewriters, this is intended as a rough guide on what to change and how it works.
+
+That said, we should probably rewrite it, so please don't take this as evidence that this is an architecture we want to keep.  For a lot of reasons:
+- To allow external extension without modifying this repository, which is currently not possible.
+- To allow scan-and-fix in one step, to reduce delay and duplicate operations.  (in large clusters, scanner can take *weeks* to complete)
+- There is a large amount of loose stringly-typed casting / binding spread across many files.
+
+Files in this doc are linked by relative paths, which works on GitHub and in most IDEs.
+Sadly, links to specific lines and repo-relative `/path/to/file` support is widely varying, so it should be avoided.
+
+# Overall behavior
+
+At the highest level, keep in mind:
+- Invariants define things to check and fix, and are defined 1:1 with their Collections but do not share names directly
+  - These are what actually check if X is bad, and do the things necessary to fix it
+  - Each handles only one source/type for their primary data, e.g. concrete vs current vs timer
+  - Invariant Managers are just N invariants that are run in a list.  Note that they are _not_ related to invariant collections.
+- Scanner and Fixer can be enabled for per data type (separate workflows) and per Collection (via config only)
```

Comment:
not sure sure i understand your point with this comment? Perhaps you could use an example? I vaguely recall there being a scanner workflow for concrete and current records

Reply:
there are three scanner workflows and three fixer workflows.
all three of each are nearly the same, but they operate on different types and have different workflow names/types.  e.g.:
```go
ConcreteExecutionsScannerWFTypeName   = "cadence-sys-executions-scanner-workflow
```

within each of those, the invariants that are run are individually enable/disable-able.

but yea, gonna reword pretty much all of this 👍